tests/func/remote/test_local.py

```diff
@@ -0,0 +1,34 @@
+import os
+
+import mock
+
+import dvc
+from dvc.main import main
+from dvc.remote.local import RemoteLOCAL
+
+
+def test_should_not_fail_add_on_unpacked_dir_creation_exception(
+    dvc_repo, repo_dir
+):
+
+    with mock.patch.object(
+        dvc.remote.local.RemoteLOCAL,
+        "_create_unpacked_dir",
+        side_effect=Exception,
```

Comment:
Ok, so looks like your test doesn't work? `Exception` should break everything. And you are catching `DvcException` in your code, so this test looks like a false-positive.